subsys/bluetooth/audio/bap_broadcast_assistant.c

```diff
@@ -384,7 +384,9 @@ static void long_bap_read(struct bt_conn *conn, uint16_t handle)
 
 static void delayed_bap_read_handler(struct k_work *work)
 {
-	long_bap_read(broadcast_assistant.conn, broadcast_assistant.long_read_handle);
+	if (broadcast_assistant.conn != NULL) {
```

Comment:
can we move this check before task scheduling ?

Reply:
Not fully. 

The `broadcast_assistant.conn` may be accessed and modified by multiple thread I believe. 

We could move the check to `long_bap_read` if you prefer

Luckily there are significant effort going on to reduce the number of threads in BT, which should make it better in the future

Reply:
I did not describe clear enough.

I was thinking if we can move this check to the function long_bap_read()

just before 

err = k_work_reschedule(&broadcast_assistant.bap_read_work, ....);

which avoids a k_work_reschedule when broadcast_assistant.conn == NULL

Reply:
We can move it to the function, yes, but it would need to be at the very top. 

It doesn't make sense to do anything in the function if `conn == NULL`, and similar it does not make sense to call the function when you know that `conn == NULL`, which is why I guarded the function call

Reply:
I thought we agree with the better solution if conn == NULL, long_bap_read()  will return at the very top instead of rescheduling the task ...  :-)